libraries/AP_Scripting/AP_Scripting_SerialAccess.cpp

```diff
@@ -24,9 +24,9 @@ void AP_Scripting_SerialAccess::begin(uint32_t baud)
     }
 }
 
-size_t AP_Scripting_SerialAccess::write(uint8_t c)
+int32_t AP_Scripting_SerialAccess::write(uint8_t c)
 {
-    return write(&c, 1);
+    return (int32_t)write(&c, 1); // return value will be 0 or 1
```

Comment:
Read returns bool, why not here too?

Reply:
Would the Lua result also be a bool? Not a bad idea, but Lua bools do not decay to int so it would be harder to e.g. keep a total of bytes sent.

Reply:
On the other hand, in Lua, 0 is truthy so it is probably best to do that so that `if port:write(c) then` works as expected.